### ISSUE TITLE
twisted/conch/test/test_agent.py always skips iosim tests

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#7977]
|<img alt="edorum2's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">| edorum2 reported|
|-|-|
|Trac ID|trac#7977|
|Type|enhancement|
|Created|2015-07-23 10:23:49Z|
|Branch|https://github.com/twisted/twisted/tree/conch-test_agent-iosim-skip-7977|

The iosim tests in twisted/conch/test/test_agent.py are always skipped, since a reactor would never implement ITLSTransport. 
But these tests does not need TLS support to work, so they could always use iosim and would pass. 

This bug is based on a fix for an issue described in [#7015](https://github.com/twisted/twisted/issues/7015).  

Attachments:

* [7977-1.diff](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/af5/af58357e617142f777c232d0b9930708b18d879c/a7e1d150b8d2ba7c5655878969544acd98f25721.diff) (682 bytes) - added by edorum2 on 2015-07-23 10:26:58Z - 

<details><summary>Searchable metadata</summary>

```
trac-id__7977 7977
type__enhancement enhancement
reporter__edorum2 edorum2
priority__normal normal
milestone__ 
branch__branches_conch_test_agent_iosim_skip_7977 branches/conch-test_agent-iosim-skip-7977
branch_author__adiroiban adiroiban
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1437647029963634 1437647029963634
changetime__1437653621169116 1437653621169116
version__None None
owner__edorum2 edorum2

```
</details>

**Comment by twisted-trac** [comment]
|<img alt="edorum2's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|edorum2 removed owner|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_2"></a>|@adiroiban commented|
|-|-|

(In [45296]) Branching to conch-test_agent-iosim-skip-7977.

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_3"></a>|@adiroiban set owner to edorum2|
|-|-|

Changes looks good. Thanks!

OS X failures are not related to this patch.

Will merge soon.

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_4"></a>|@adiroiban set status to `closed`|
|-|-|

(In [45298]) Merge conch-test_agent-iosim-skip-7977: Enable iosim tests in twisted.conch.test.test_agent.

Author: edorum2
Reviewer: adiroiban
Fixes: [#7977](https://github.com/twisted/twisted/issues/7977)
